nan.h

```diff
@@ -637,18 +645,76 @@ NAN_INLINE _NanWeakCallbackInfo<T, P>* NanMakeWeakPersistent(
     return _NAN_ERROR(v8::Exception::TypeError, errmsg);
   }
 
+  NAN_INLINE v8::Local<v8::Value> NanTypeError(v8::Handle<v8::String> errmsg) {
+    return v8::Exception::TypeError(errmsg);
+  }
+
   NAN_INLINE void NanThrowTypeError(const char* errmsg) {
     _NAN_THROW_ERROR(v8::Exception::TypeError, errmsg);
   }
 
+  NAN_INLINE void NanThrowTypeError(v8::Handle<v8::String> errmsg) {
+    NanScope();
+    v8::Isolate::GetCurrent()->ThrowException(
+        v8::Exception::TypeError(errmsg));
+  }
+
+  NAN_INLINE v8::Local<v8::Value> NanSyntaxError(const char* errmsg) {
+    return _NAN_ERROR(v8::Exception::SyntaxError, errmsg);
+  }
+
+  NAN_INLINE
+  v8::Local<v8::Value> NanSyntaxError(v8::Handle<v8::String> errmsg) {
+    return v8::Exception::SyntaxError(errmsg);
+  }
+
+  NAN_INLINE void NanThrowSyntaxError(const char* errmsg) {
+    _NAN_THROW_ERROR(v8::Exception::SyntaxError, errmsg);
+  }
+
+  NAN_INLINE void NanThrowSyntaxError(v8::Handle<v8::String> errmsg) {
+    NanScope();
+    v8::Isolate::GetCurrent()->ThrowException(
+        v8::Exception::SyntaxError(errmsg));
+  }
+
+  NAN_INLINE v8::Local<v8::Value> NanReferenceError(const char* errmsg) {
+    return _NAN_ERROR(v8::Exception::ReferenceError, errmsg);
+  }
+
+  NAN_INLINE
+  v8::Local<v8::Value> NanReferenceError(v8::Handle<v8::String> errmsg) {
+    return v8::Exception::ReferenceError(errmsg);
+  }
+
+  NAN_INLINE void NanThrowReferenceError(const char* errmsg) {
+    _NAN_THROW_ERROR(v8::Exception::ReferenceError, errmsg);
+  }
+
+  NAN_INLINE void NanThrowReferenceError(v8::Handle<v8::String> errmsg) {
+    NanScope();
+    v8::Isolate::GetCurrent()->ThrowException(
+        v8::Exception::ReferenceError(errmsg));
+  }
+
```

Comment:
I would probably use an x-macro for this.  There is a lot of duplication with only minor changes in functionality.  It makes it somewhat hard to eyeball what the differences are and if they are correct.

An x-macro would probably also let you get rid of the _NAN_THROW_ERROR macro.